transmit-lib/src/cmr/transmit/http_helper.clj

```diff
@@ -105,6 +121,35 @@
                             http-options)))]
     (response-handler request response)))
 
+(defn request-with-returned-headers
+  "Makes an HTTP request with the given options and parses the response. The arguments in the
+  options map have the following effects.
+
+  * :url-fn - A function taking a transmit connection and returning the URL to use.
+  * :method - the HTTP method. :get, :post, etc.
+  * :raw? - indicates whether the raw HTTP response (as returned by http-response->raw-response ) is
+  desired. Defaults to false.
+  * :use-system-token? - indicates if the ECHO system token should be put in the header
+  * :http-options - a map of additional HTTP options to send to the clj-http.client/request function.
+  * :response-handler - a function to handle the response. Defaults to default-response-handler"
+  [context app-name {:keys [url-fn method http-options response-handler use-system-token?] :as request}]
+  (let [conn (config/context->app-connection context app-name)
+        response-handler (or response-handler default-response-handler)
+        connection-params (config/conn-params conn)
+        ;; Validate that a connection manager is always present. This can cause poor performance if not.
+        _ (when-not (:connection-manager connection-params)
+            (errors/internal-error! (format "No connection manager created for [%s] in current application" app-name)))
+        response (http-response->raw-response-with-headers
+                   (client/request
+                     (merge connection-params
+                            {:url (url-fn conn)
+                             :method method
+                             :throw-exceptions false}
+                            (when use-system-token?
+                              {:headers {config/token-header (config/echo-system-token)}})
+                            http-options)))]
+    response))
```

Comment:
why can't we use the original request function on line 79?  Its the exact same function, except that it calls the raw-response-with-headers that I am questioning why we need it as well as the return of response, which I think is incorrect.  On line 151 I think it should be (response-handler request response).